assets/js/script.js

```diff
@@ -0,0 +1,91 @@
+var grid = Array(100).fill(0)
+
+//store snakes and ladders keys. jump refers to change of position
+var jumps = {
+  1: 38,
+  4: 14,
+  9: 31,
+  17: 7,
+  21: 42,
+  28: 84,
+  51: 67,
+  54: 34,
+  62: 19,
+  64: 60,
+  72: 91,
+  80: 99,
+  87: 36,
+  93: 73,
+  95: 75,
+  98: 79,
+}
+
+function getJump(rollValue) {
+  if (jumps[rollValue]) {
+    //  console.log(jumps[rollValue]);
+    return jumps[rollValue]
+  } else return 0
+
+}
+
+var player = 1
+var move = 0
+
+function whoWon() {
+  if (grid[100] > 0) {
+    $("#winner").text(`Winner is : ${grid[100]}`)
+    return grid[100]
+  }
+}
+
+//function set grid
+function playTurn() {
+  // console.log('PLAYER START:', player);
+  //get last move of this player, last index of player. Set to 1 at start of game (index will be -1).
+  var lastMove = grid.lastIndexOf(player) < 1 ? 0 : grid.lastIndexOf(player)
+  //get random value 1-6
+  var roll = Math.floor(Math.random() * 6) + 1
+  // console.log('rollValue', roll);
+  var nextPos = lastMove + roll
+  // console.log('jump', getJump(nextPos));
+  move = getJump(nextPos) === 0 ? nextPos : getJump(nextPos)
```

Comment:
When the character move, the grid is updated but the old position is not deleted. Hence if the character takes a snake down. On the next character's turn, it takes the character's furthest traveled position due to grid.lastIndexOf() and starts the next position move from there, this invalidates the role of snakes in the game.

Reply:
agree. grid should have only one 1 and 2.